### ISSUE TITLE
LiveAnnouncer does not work when made inside a mat-dialog

### ISSUE THREAD

**Issue by dlpx-gli** [angular/components#22733]
#### Reproduction

Use StackBlitz to reproduce your issue: 
https://stackblitz.com/edit/components-issue-fksfhp?file=src/app/dialog-content-example.ts

Steps to reproduce:
1. Open screen reader, I used mac voiceover. Everything below are done via screen reader.
2. Click on the "Open dialog" button to open dialog
3. Inside the dialog, click the "test" button which would make a live announcement.
4. It says "press test, button". No announcement made. If I open the inspector, the `cdk-live-announcer-element` region is correctly populated with the announcement text despite no actual announcement. In addition, if I move this div inside `<mat-dialog-container>` it would work but then this element becomes reacheable via screen reader.
 
 
#### Expected Behavior

What behavior were you expecting to see?

Make the announcement.

#### Actual Behavior

What behavior did you actually see?

No announcement even though `cdk-live-announcer-element` did get populated.

#### Environment

- Angular: "10.2.4"
- CDK/Material: "10.2.7"
- Browser(s): Chrome Version 90.0.4430.93
- Operating System (e.g. Windows, macOS, Ubuntu): macOS mojave 10.14.6

**Comment by dlpx-gli** [comment]
The problem seems to be specific to chrome. It works fine for firefox.

**Comment by crisbeto** [comment]
What screen reader is this on? It worked correctly for me with NVDA on Chrome.

**Comment by dlpx-gli** [comment]
> What screen reader is this on? It worked correctly for me with NVDA on Chrome.

This was the voiceover on macOS mojave 10.14.6

**Comment by crisbeto** [comment]
If that's the case, I suspect that there may be issues with the `LiveAnnouncer` outside of a dialog as well, because the two are basically the same. I'll mark this as "needs investigation" so that we can verify it against VoiceOver.

**Comment by zelliott** [comment]
Without having done any investigation into this, it could be due to the dialog's focus trap implementation. I wonder if the `LiveAnnouncer` element is having `aria-hidden="true"` applied, or if the existence of `aria-modal="true"` on the dialog is preventing SRs from announcing the `LiveAnnouncer` element.

**Comment by zarend** [comment]
Reopening as this appears to have regressed. I can reproduce this issue again on Chrome with VoiceOver.

1. With Screen Reader enabled open "/dialog" on the dev-app.
2. Click "Open dialog with template context"
3. ("Order printer ink refills." form appears)
4. Click the "What Color" form field
5. (List of options appears and blank option is focused)

Expected Behavior:

Screen reader announces the focused option, which is the "blank" option.

Actual Behavior

Screen reader makes no announcement, and I am unable to navigate to the popup options.

 - Chrome Version 118.0.5993.70 (Official Build) (arm64)
 - Macos 13.6 (22G120)
 - VoiceOver

I reproduced using Select component, which has the same fix as LiveAnnouncer to apply the aria-owns. I can see that the aria-owns is being set correctly on the modal, so I'm not sure what regressed. (cc @crisbeto )
